src/meshes/p4est_icosahedron_quads.jl

```diff
@@ -0,0 +1,774 @@
+                v4 = local_mapping(x_0, y_1, v1_quad, v2_quad, v3_quad, v4_quad, radius)
+
+                for j in eachindex(nodes), i in eachindex(nodes)
+                    # node_coordinates are the mapped reference node coordinates
+                    node_coordinates[:, i, j, tree] .= local_mapping(nodes[i], nodes[j],
+                                                                     v1, v2, v3, v4,
+                                                                     radius)
+                end
+            end
+        end
+    end
+end
+
+# Icosahedron vertices
+#            5
+#           ,'|.     
+#         ,'  | `.   
+#       ,'    |   `. 
+#    6,'_____1|_____`.4
+#     \      /\      /
+#      \    /  \    /
+#       \  /    \  /
+#       2\/______3/______4_______5_______6_______2     
+#         \      /\      /\      /\      /\      /\    
+#          \    /  \    /  \    /  \    /  \    /  \   
+#           \  /    \  /    \  /    \  /    \  /    \  
+#            \/_______/_______/_______/_______/______\7
+#            7       8       9      10      11\      /\   
+#                                           /  \    /  \  
+#                                          /    \  /    \ 
+#                                         /______\/______\
+#                                       10 .     |12    ,'8
+#                                           `.   |    ,'  
+#                                             `. |  ,'    
+#                                               `|,'
+#                                                9
+function calc_node_coordinates_icosahedron_vertices(radius; RealT = Float64)
+    vertices = Array{RealT, 2}(undef, 3, 12)
+
+    vertices[:, 1] = [0, 0, 1]
+    vertices[:, 2] = [
+        -sqrt(1 / 10 * (5 - sqrt(5))),
+        -1 / 2 - 1 / (2 * sqrt(5)),
+        1 / sqrt(5)
+    ]
+    vertices[:, 3] = [
+        sqrt(1 / 10 * (5 - sqrt(5))),
+        -1 / 2 - 1 / (2 * sqrt(5)),
+        1 / sqrt(5)
+    ]
+    vertices[:, 4] = [
+        sqrt(1 / 10 * (5 + sqrt(5))),
+        1 / 2 - 1 / (2 * sqrt(5)),
+        1 / sqrt(5)
+    ]
+    vertices[:, 5] = [0, 2 / sqrt(5), 1 / sqrt(5)]
+    vertices[:, 6] = [
+        -sqrt(1 / 10 * (5 + sqrt(5))),
+        1 / 2 - 1 / (2 * sqrt(5)),
+        1 / sqrt(5)
+    ]
+    vertices[:, 7] = [0, -2 / sqrt(5), -1 / sqrt(5)]
+    vertices[:, 8] = [
+        sqrt(1 / 10 * (5 + sqrt(5))),
+        1 / (2 * sqrt(5)) - 1 / 2,
+        -1 / sqrt(5)
+    ]
+    vertices[:, 9] = [
+        sqrt(1 / 10 * (5 - sqrt(5))),
+        1 / 2 + 1 / (2 * sqrt(5)),
+        -1 / sqrt(5)
+    ]
+    vertices[:, 10] = [
+        -sqrt(1 / 10 * (5 - sqrt(5))),
+        1 / 2 + 1 / (2 * sqrt(5)),
+        -1 / sqrt(5)
+    ]
+    vertices[:, 11] = [
+        -sqrt(1 / 10 * (5 + sqrt(5))),
+        1 / (2 * sqrt(5)) - 1 / 2,
+        -1 / sqrt(5)
+    ]
+    vertices[:, 12] = [0, 0, -1]
+
+    return vertices * radius
+end
+
+function icosahedron_triangle_vertices_idx(triangle)
```

Comment:
Can this not just be extracted from a matrix? That is more like how things are done, for example, in StartUpDG.

Reply:
In cd6bffab3dc57b4232c06e12549507702386a8d5, I replaced the two functions that defined index maps with constant tuples. I think that this solution is cleaner and allows for fast and efficient access to the index maps.